ui/apps/platform/src/hooks/useURLPagination.ts

```diff
@@ -8,12 +8,15 @@ type UseURLPaginationResult = {
     setPerPage: (perPage: number) => void;
 };
 
+function safeNumber(val: unknown, defaultVal: number) {
+    const parsed = Number(val);
+
+    return Number.isNaN(parsed) ? defaultVal : parsed;
```

Comment:
For `page` and `perPage` you might consider the following positive condition:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger

Reply:
While we're at it, we may want to ensure it's also an integer > 1

```suggestion
    return Number.isSafeInteger(parsed) && parsed > 0 ? parsed : defaultVal;
```